code/game/machinery/metaldetector.dm

```diff
@@ -219,24 +197,9 @@
 
 		var/dat = "<h4>"
 
-		if(idmode == 0)
-			dat += "Citizens must carry ID: <A href='?src=\ref[src];idmode=1'>Turn On</A><BR><BR>"
-		else
-			dat += "Citizens must carry ID: <A href='?src=\ref[src];idmode=1'>Turn Off</A><BR><BR>"
-
-
-		if(scanmode == 0)
-			dat += "Intrusive Scan: <A href='?src=\ref[src];scanmode=1'>Turn On</A><BR><BR>"
-		else
-			dat += "Intrusive Scan: <A href='?src=\ref[src];scanmode=1'>Turn Off</A><BR><BR>"
-
-
-
-		if(senset == 0)
-			dat += "DeMil Alerts: <A href='?src=\ref[src];senmode=1'>Turn On</A><BR><BR>"
-		else
-			dat += "DeMil Alerts: <A href='?src=\ref[src];senmode=1'>Turn Off</A><BR><BR>"
-
+		dat += "Citizens must carry ID: <A href='?src=\ref[src];action=idmode'>Turn [idmode ? "Off" : "On"]</A><BR><BR>"
```

Comment:
Convert these to {""} instead of adding all of them to strings like this. Strings are slow in BYOND so it's cheaper to do it all at once.